cli/Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -40,21 +63,147 @@ public enum AuthenticationToken: CustomStringConvertible, Equatable {
     public var description: String {
         switch self {
         case .user:
-            return "tuist user token: \(value)"
+            return "User token: \(value)"
         case let .project(token):
-            return "tuist project token: \(token)"
+            return "Project token: \(token)"
         }
     }
 }
 
+#if canImport(TuistSupport)
+    /// Actor responsible for serializing file system lock operations to prevent race conditions
+    private actor FileSystemLockActor {
```

Comment:
I'm using an actor here because there are highly concurrent workflows like `cache` that might cause a race condition accessing the lockfile.